cunumeric/deferred.py

```diff
@@ -708,7 +708,11 @@ def _broadcast(self, shape: NdShape) -> Any:
 
         for dim in range(len(shape)):
             if result.shape[dim] != shape[dim]:
-                assert result.shape[dim] == 1
+                if result.shape[dim] != 1:
+                    raise ValueError(
+                        f"Shape did not match along dimension {dim}"
```

Comment:
Leave a space at the end of the string here, otherwise when they get joined it will look like this: "... dimension 3and the ..."

Reply:
Fixed. Thanks. @manopapad After this review, can I merge the change?